build/helper.py

```diff
@@ -179,28 +198,23 @@ def _get_output_param_return_snippet(output_parameter):
     '''Returns the snippet for returning a single output parameter from a Session method, i.e. "reading_ctype.value"'''
     assert output_parameter['direction'] == 'out', pp.pformat(output_parameter)
     if output_parameter['is_buffer']:
-        if output_parameter['type'] == 'ViChar':
+        if output_parameter['type'] == 'ViChar' or output_parameter['type'] == 'ViString':
             snippet = output_parameter['ctypes_variable_name'] + '.value.decode("ascii")'
         else:
             # TODO(marcoskirsch): I don't like calling camelcase_to_snakecase here, it relies on contract that parameter name where the size is stored was created with that function.
-            # TODO(texasaggie97): Not implementing ivi-dance so replace with 0
-            size_name = camelcase_to_snakecase(output_parameter['size'])
-            if size_name == 'ivi-dance':
-                size_name = '0'
-            snippet = '[' + output_parameter['ctypes_variable_name'] + '[i].value for i in range(' + size_name + ')]'
+            snippet = '[' + output_parameter['ctypes_variable_name'] + '[i].value for i in range(' + camelcase_to_snakecase(output_parameter['size']) + ')]'
     else:
         snippet = output_parameter['ctypes_variable_name'] + '.value'
 
     return snippet
 
-def get_method_return_snippet(output_parameters):
+def get_method_return_snippet(parameters):
     '''Returns a string suitable to use as the return argument of a Session method, i.e. "return reading_ctype.value"'''
-    if len(output_parameters) == 0:
-        return 'return'
     snippets = []
-    for x in output_parameters:
-        snippets.append(_get_output_param_return_snippet(x))
-    return 'return ' + ', '.join(snippets)
+    for x in parameters:
+        if x['direction'] == 'out' or x['ivi-dance']:
```

Comment:
Theoretically yes. I can remove the or.

This brings up a question though. Should we run some sort of validation on the metadata? To ensure there aren't places that don't meet these type of assumptions?